App/Services/Fetcher.cs

```diff
@@ -279,21 +279,333 @@ public async Task<Article> GetArticle(string articleId)
+            await _generalDB.UpdateFeed(feed);
+            string token = await SecureStorage.GetAsync(AppConstant.NotificationToken);
+
+            if (string.IsNullOrEmpty(token))
+                return feed;
+            
+            await SubscribeToFeed(feed.MongoID, token);
+
+            return res.Data;
+        }
+
+        catch (Exception ex)
+        {
+#if DEBUG
+            Debug.WriteLine(ex);
+#else
+            SentrySdk.CaptureException(ex);
+#endif
+            return null;
+        }
+    }
+
+    /// <summary>
+    /// Create a feed
+    /// </summary>
+    /// <param name="feed">ID of the feed concerned by the subscription</param>
+    public async Task<Feed> UpdateFeed(Feed feed)
+    {
+        try
+        {
+            string name = feed.Title;
+            string keyword = feed.Keywords;
+            string id = feed.MongoID;
+            await _generalDB.UpdateFeed(feed);
+
+            if (string.IsNullOrEmpty(id) || Connectivity.Current.NetworkAccess != NetworkAccess.Internet)
+                return feed;
+
+            Dictionary<string, string> rqHeaders = new();
+            if (UserData != null)
+                rqHeaders.Add("Authorization", $"{await SecureStorage.GetAsync(nameof(Session.TokenType))} {await SecureStorage.GetAsync(nameof(Session.AccessToken))}");
+
+
+            var res = await this.WebService.Put<FeedResponse>(controller: "feeds",
+                                           action: "custom/update",
+                                           parameters: new Dictionary<string, string>
+                                           {
+                                                { nameof(name), name },
+                                                { nameof(keyword), keyword },
+                                                { nameof(id), id },
+                                            },
+                                           singleUseHeaders: rqHeaders.Count > 0? rqHeaders: null,
+                                           unSuccessCallback: e => _ = HandleHttpException(e));
+
+
+            return res.Data;
+        }
+
+        catch (Exception ex)
+        {
+#if DEBUG
+            Debug.WriteLine(ex);
+#else
+            SentrySdk.CaptureException(ex);
+#endif
+            return null;
+        }
+    }
+
+    /// <summary>
+    /// Subcribe to a feed notification subscription
+    /// </summary>
+    /// <param name="feedID">ID of the feed concerned by the subscription</param>
+    /// <param name="token">notification token link to the NE of the user</param>
+    /// <returns>retrun the article</returns>
+    public async Task UnsubscribeToFeed(string feedID, string token)
+    {
+        try
+        {
+            Dictionary<string, string> rqHeaders = new();
+            if (UserData != null)
+                rqHeaders.Add("Authorization", $"{await SecureStorage.GetAsync(nameof(Session.TokenType))} {await SecureStorage.GetAsync(nameof(Session.AccessToken))}");
```

Comment:
This is repeated on purpose, we'd need to create a method that initialise all that before each call